amm/interp/src/main/scala/ammonite/interp/AmmonitePlugin.scala

```diff
@@ -94,7 +98,61 @@ object AmmonitePlugin{
       !ignoredNames(sym.name.decoded)
     }
 
-    val stats = unit.body.children.last.children.last.asInstanceOf[g.ImplDef].impl.body
+    val stats = {
+      val nestingLevel = userCodeNestingLevel
+      assert(nestingLevel >= 0)
+      (0 until nestingLevel).foldLeft(unit.body.children.last.children)((res, _) =>
+        res.last.asInstanceOf[g.ImplDef].impl.body
+      )
+    }
+
+    val uses =
+      if (userCodeNestingLevel <= 1)
+        Map.empty[String, Seq[String]]
```

Comment:
Let's add a comment here saying we *could* support computing `uses` (which we should probably name something longer such as `usedEarlierDefinitions`) for `userCodeNestingLevel == 1`, but explicitly choose not to do so because it's currently only used in the class-based-wrapper which has `userCodeNestingLevel == 2`

Reply:
Ok